tests/test_geometry_3d.h

```diff
@@ -0,0 +1,422 @@
+/* Copyright (c) 2014-2021 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef TEST_3D_GEOMETRY_H
+#define TEST_3D_GEOMETRY_H
+
+#include "core/math/geometry_3d.h"
+#include "core/math/plane.h"
+#include "core/math/random_number_generator.h"
+#include "core/math/vector3.h"
+#include "tests/test_macros.h"
+#include "vector"
+namespace Test3DGeometry {
+/// <summary>
+/// Static maths function tests
+/// </summary>
+TEST_CASE("[Geometry3D] get_closest_points_between_segments") {
+	struct Case {
+		Vector3 Point1, Point2, Point3, Point4;
+		Vector3 Got1, Got2;
+		Vector3 Want1, Want2;
+		Case(){};
+		Case(Vector3 point1, Vector3 point2, Vector3 point3, Vector3 point4, Vector3 want1, Vector3 want2) :
+				Point1(point1), Point2(point2), Point3(point3), Point4(point4), Want1(want1), Want2(want2){};
+	};
+	Vector<Case> tt;
+	tt.push_back(Case(Vector3(1, -1, 1), Vector3(1, 1, -1), Vector3(-1, -2, -1), Vector3(-1, 1, 1), Vector3(1, 0, 0), Vector3(-1, 0, 0)));
+	for (int i = 0; i < tt.size(); ++i) {
+		Case currentCase = tt[i];
+		Geometry3D::get_closest_points_between_segments(currentCase.Point1, currentCase.Point2, currentCase.Point3, currentCase.Point4, currentCase.Got1, currentCase.Got2);
+
+		CHECK(Vector3(round(currentCase.Got1.x), round(currentCase.Got1.y), round(currentCase.Got1.z)) == currentCase.Want1);
+		CHECK(Vector3(round(currentCase.Got2.x), round(currentCase.Got2.y), round(currentCase.Got2.z)) == currentCase.Want2);
```

Comment:
Is rounding needed because of floating point error? You can use `Vector3.is_equal_approx()` for this.

Reply:
Fixed this too